ext/http/lib.rs

```diff
@@ -425,6 +428,39 @@ impl Default for HttpResponseWriter {
   }
 }
 
+struct BodyCompressedHolder {
+  pub writer: Pin<Box<dyn tokio::io::AsyncWrite>>,
+  pub shutdown_handle: ShutdownHandle,
+}
+
+struct BodyUncompressedSender {
+  sender: Option<hyper::body::Sender>,
```

Comment:
Answer the question below first. But if you decide that you definitely want a boolean, drop the `Option` wrapper.